### ISSUE TITLE
Does not appear to be a git repository

### ISSUE THREAD

**Issue by moronicgeek** [piku/piku#195]
Hi there, 
I have followed the process below:

- Install Raspberry OS on a card
- rename pi
- activate password for root
- PermitLogin yes for root
- bootstrap the server with ./piku-bootstrap root@mypi.local (Runs successfully)
- install the playbooks : nodeenv and nodejs playbooks
- Try to install node js example app in examples 
- Run : git remote add piku piku@your_server:nodeapp
- Run : git push piku master

And then i get this error:

```
fatal: 'nodeapp' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

What i have noticed is that i can still log into the piku user however and i expected it to run the piku.py script. 

More info: 
I was able to get the deployments running before i wiped the card and ran the ansible playbook by running a mixture of commands from the install.md and the raspbian readme's. 

So once i have this working i would love to put my hand up to help restructure the readme to be more readable and following a set of commands to get up and running quickly

**Comment by chr15m** [comment]
`git remote add piku piku@your_server:nodeapp`

This should probably be: `git remote add piku piku@mypi.local:nodeapp` yes?

> What i have noticed is that i can still log into the piku user however and i expected it to run the piku.py script.

It sounds like it has not set up the SSH hook corrrectly.

> So once i have this working i would love to put my hand up to help restructure the readme to be more readable and following a set of commands to get up and running quickly

Sounds great. If you can help me debug the bootstrap script so it's a one-run install that would be very good.

**Comment by moronicgeek** [comment]
Great i will havea look. Would you mind explaining how to manually setup the ssh hook? 
Perhaps that would give me a start at where to look at ?

**Comment by chr15m** [comment]
@moronicgeek I think you should be able to do this to set up the SSH hook:

`python3 ~/piku.py setup:ssh /path/to/ssh-key.pub`

**Comment by timhuebener** [comment]
I have the same problem 
- download piku-bootstrap on server
- run `./piku-bootstrap install` as sudo
- give the piku user a new password on server (I did this because I was prompted a password and the `!` password in the piku playbook did not work)
- install piku client on local machine
- add piku remote to git repro on local machine `git remote add piku piku@piku-server:app-name` 
- commit on local machine
- run `git push piku master`

and then I get `fatal: 'app-name' does not appear to be a git repository

I then tried this `python3 ~/piku.py setup:ssh /path/to/ssh-key.pub` as the piku user on the server. This returned
> Adding key 'SSHKEY...'.

but this did not fix it unfortunately.

**Comment by chr15m** [comment]
After you added the SSH key are you able to SSH to `piku@piku-server` without entering a password?

**Comment by timhuebener** [comment]
no it always asks for a password

**Comment by chr15m** [comment]
@timhuebener in that case I think it's likely there is an issue with your SSH keys setup. You need to make sure your key is in `.ssh/authorized_keys` for the `piku` user on the server and that the `.ssh` dir and the file itself have the correct permissions and that the server is configured to allow user keys.

You should be able to find more info in `/var/log/auth.log`.

**Comment by timhuebener** [comment]
@chr15m thanks for the tip! 

my keys were in the `.ssh/authorized_keys`. The problem was, that I had to set `PasswordAuthentication no` in `/etc/ssh/sshd_config`

Then another problem I ran into was that `/usr/lib/python3/dist-packages/nodeenv.py:503` `KeyError: 'aarch64'`. RasberryPi4 returnes `aarch64` when executing `platform.machine()` which means that `postfix_arch = arches[platform.machine()]` failed. That was easily fixed by adding `aarch64: 'arm64'`

Next, I got this error 
```
remote: File "/home/piku/piku.py", line 584, in deploy_node
remote: unlink(node_path_tmp)
remote: IsADirectoryError: [Errno 21] Is a directory: '/home/piku/.piku/apps/accounting/node_modules'
```
I think I have solved this by replacing `unlink` with `rmdir` ... at least I don't get an error anymore 

Now I'm stuck at this error 
```
remote: File "/home/piku/piku.py", line 744, in spawn_app
remote: symlink(join(ACME_ROOT, domain), join(ACME_WWW, app))
remote: FileNotFoundError: [Errno 2] No such file or directory: '/home/piku/.acme.sh/<URL>' -> '/home/piku/.piku/acme/<app-name>'
```

**Comment by rcarmo** [comment]
I’ve been reading through this and I can’t fathom it. I have Pi 4s running Ubuntu, Raspbian (both current and last release) running Node apps, and can’t reproduce it (but I run Nodesource packages and only bare npm).

**Comment by chr15m** [comment]
@timhuebener if you SSH in as root and then `su piku` and then run `python` what version does it output?

**Comment by timhuebener** [comment]
@rcarmo I installed Raspbian 32Bit on my Pi4 and now it works as well. Though, I had to change `/usr/lib/python3/dist-packages/nodeenv.py:503 KeyError: 'armv7l'` this again. (just for armv7l now) 

But now it everything works 🎉 

before I was using ubuntu server 64bit. So apparently that does not really work well with Piku 😄 
@chr15m unfortunately I can't do that anymore ... but it was ubuntu server 20.4LTS so probably a python3 version
